tools/istio-iptables/main.go

```diff
@@ -62,140 +104,82 @@ func separateV4V6(cidrList string) (NetworkRange, NetworkRange, error) {
 	return ipv4Ranges, ipv6Ranges, nil
 }
 
-func run(args []string, flagSet *flag.FlagSet) {
-
+func constructConfigFromFlags(args []string, flagSet *flag.FlagSet) *Config {
+	config := &Config{}
 	proxyUID := ""
 	proxyGID := ""
+	kubevirtInterfaces := ""
+
 	inboundInterceptionMode := env.RegisterStringVar("ISTIO_INBOUND_INTERCEPTION_MODE", "", "").Get()
-	inboundTProxyMark := env.RegisterStringVar("ISTIO_INBOUND_TPROXY_MARK", "1337", "").Get()
-	inboundTProxyRouteTable := env.RegisterStringVar("ISTIO_INBOUND_TPROXY_ROUTE_TABLE", "133", "").Get()
+	config.InboundTProxyMark = env.RegisterStringVar("ISTIO_INBOUND_TPROXY_MARK", "1337", "").Get()
+	config.InboundTProxyRouteTable = env.RegisterStringVar("ISTIOINBOUND_TPROXY_ROUTE_TABLE", "133", "").Get()
 	inboundPortsInclude := env.RegisterStringVar("ISTIO_INBOUND_PORTS", "", "").Get()
 	inboundPortsExclude := env.RegisterStringVar("ISTIO_LOCAL_EXCLUDE_PORTS", "", "").Get()
 	outboundIPRangesInclude := env.RegisterStringVar("ISTIO_SERVICE_CIDR", "", "").Get()
 	outboundIPRangesExclude := env.RegisterStringVar("ISTIO_SERVICE_EXCLUDE_CIDR", "", "").Get()
 	outboundPortsExclude := env.RegisterStringVar("ISTIO_LOCAL_OUTBOUND_PORTS_EXCLUDE", "", "").Get()
-	kubevirtInterfaces := ""
-	var enableInboundIPv6s net.IP
-
 	proxyPort := env.RegisterStringVar("ENVOY_PORT", "15001", "").Get()
 	inboundCapturePort := env.RegisterStringVar("INBOUND_CAPTURE_PORT", "15006", "").Get()
-	flagSet.StringVar(&proxyPort, "p", proxyPort,
+
+	flagSet.StringVar(&config.ProxyPort, "p", proxyPort,
```

Comment:
This whole thing should move to `cobra` to get help output, etc.
At least, as a first step, move the definition of flags into a `func init()`.

Reply:
yeah.. will add as a separate commit..

Reply:
Moved to cobra and viper.